crates/lib3h/src/engine/ghost_engine_wrapper.rs

```diff
@@ -0,0 +1,323 @@
+use crate::engine::ghost_engine::{CoreRequestContext, GhostEngineParentWrapper};
+use detach::Detach;
+use lib3h_ghost_actor::*;
+use lib3h_protocol::{
+    data_types::GenericResultData, error::Lib3hProtocolResult, protocol::*, protocol_client::*,
+    protocol_server::*, DidWork,
+};
+
+/// A wrapper for talking to lib3h using the legacy Lib3hClient/Server enums
+#[allow(dead_code)]
+struct LegacyLib3h<Engine, EngineError: 'static>
+where
+    Engine: GhostActor<
+        Lib3hToClient,
+        Lib3hToClientResponse,
+        ClientToLib3h,
+        ClientToLib3hResponse,
+        EngineError,
+    >,
+{
+    lib3h: Detach<GhostEngineParentWrapper<LegacyLib3h<Engine, EngineError>, Engine, EngineError>>,
```

Comment:
I'd rename this member `engine` or `inner_engine` as `Lib3h::LegacyLib3h.lib3h` is a bit confusing.